wallet/src/de/schildbach/wallet/ui/widget/ReceiveInfoView.kt

```diff
@@ -0,0 +1,136 @@
+import de.schildbach.wallet.WalletApplication
+import de.schildbach.wallet.ui.ReceiveActivity
+import de.schildbach.wallet.util.Qr
+import de.schildbach.wallet.util.Toast
+import de.schildbach.wallet_test.R
+import kotlinx.android.synthetic.main.receive_info_view.view.*
+import org.bitcoinj.core.Address
+import org.bitcoinj.core.Coin
+import org.bitcoinj.uri.BitcoinURI
+import org.bitcoinj.uri.BitcoinURIParseException
+import org.dash.wallet.common.Configuration
+import org.slf4j.LoggerFactory
+
+
+class ReceiveInfoView(context: Context, attrs: AttributeSet?) : ConstraintLayout(context, attrs) {
+
+    private val log = LoggerFactory.getLogger(ReceiveInfoView::class.java)
+
+    private var config: Configuration? = null
+
+    private lateinit var address: Address
+    var amount: Coin? = null
+        set(value) {
+            field = value
+            refresh()
+        }
+
+    private lateinit var paymentRequestUri: String
+    private lateinit var qrCodeBitmap: BitmapDrawable
+
+    init {
+        inflate(context, R.layout.receive_info_view, this)
+
+        val attrsArray = context.obtainStyledAttributes(attrs, R.styleable.ReceiveInfoView)
+        try {
+            val showAmountAction = attrsArray.getBoolean(R.styleable.ReceiveInfoView_ri_show_amount_action, true)
+            specify_amount_button.visibility = if (showAmountAction) View.VISIBLE else View.GONE
+            val showShareAction = attrsArray.getBoolean(R.styleable.ReceiveInfoView_ri_show_share_action, true)
+            share_button.visibility = if (showShareAction) View.VISIBLE else View.GONE
+
+            val qrPreviewScale = attrsArray.getFloat(R.styleable.ReceiveInfoView_ri_qr_code_scale, 1.0f)
+            (qr_preview.layoutParams as LayoutParams).matchConstraintPercentWidth = qrPreviewScale
+            (qr_dash_logo.layoutParams as LayoutParams).matchConstraintPercentWidth = (qrPreviewScale / 5.5f)
+
+        } finally {
+            attrsArray.recycle()
+        }
+
+        if (!isInEditMode) {
+            val walletApplication = context.applicationContext as WalletApplication
+            config = walletApplication.configuration
+
+            address_preview_pane.setOnClickListener {
+                handleCopyAddress()
+            }
+            specify_amount_button.setOnClickListener {
+                handleSpecifyAmount()
+            }
+            share_button.setOnClickListener {
+                handleShare()
+            }
+
+            refresh()
+        }
+    }
+
+    public fun refresh() {
+        refreshData()
+
+        qrCodeBitmap = BitmapDrawable(resources, Qr.bitmap(paymentRequestUri))
+        qrCodeBitmap.isFilterBitmap = false
+
+        qr_preview.setImageDrawable(qrCodeBitmap)
+        address_preview.text = address.toBase58()
+    }
+
+    private fun refreshData() {
+        val walletApplication = context.applicationContext as WalletApplication
+        address = walletApplication.wallet.currentReceiveAddress()
```

Comment:
This should use wallet.freshReceiveAddress() to prevent address reuse.

Reply:
Good find!
Fixed.